compendium/EM/test/TestEvent.cpp

```diff
@@ -0,0 +1,183 @@
+/*=============================================================================
+
+ Library: CppMicroServices
+
+ Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+ file at the top-level directory of this distribution and at
+ https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ =============================================================================*/
+
+#include <string>
+
+#include "cppmicroservices/em/EMConstants.hpp"
+#include "cppmicroservices/em/Event.hpp"
+
+#include "cppmicroservices/Any.h"
+#include "cppmicroservices/AnyMap.h"
+#include "cppmicroservices/LDAPFilter.h"
+
+#include "gtest/gtest.h"
+
+using cppmicroservices::service::em::Event;
+using cppmicroservices::service::em::EventProperties;
+namespace emc = cppmicroservices::em::Constants;
+
+namespace {
+TEST(EventTest, TestConstructionValidTopic)
```

Comment:
There should be tests that cover topics that are specified with illegal characters that should make sure that the regex being used to validate the topics behaves properly.

Reply:
Resolved